src/common/main/kotlin/tech/thatgravyboat/skyblockapi/api/remote/api/SimpleItemApi.kt

```diff
@@ -0,0 +1,181 @@
+                attribute.shardName() to attribute(attribute.id()),
+            )
+        }.toMap().let(nameCache::putAll)
+
+        RepoAPI.items().items().entries.mapNotNull { (id, element) ->
+            val components =
+                element.getPath("['components'].['minecraft:custom_name'].['text']") ?: return@mapNotNull null
+            components.asString.stripColor() to item(id)
+        }.toMap().let(nameCache::putAll)
+
+        val newCache = nameCache.mapKeys { (key) -> key.lowercase().stripColor() }
+            .entries.flatMap { (key, value) ->
+                listOf(
+                    key to value,
+                    key.sanitizeForCommandInput() to value,
+                )
+            }.distinct().toMap()
+        nameCache.clear()
+        nameCache.putAll(newCache)
+        SkyBlockAPI.trace("Cached ${nameCache.size} item names in ${start.since().toReadableTime(allowMs = true)}")
+    }
+
+    fun findIdByName(name: String) = nameCache[name.lowercase().stripColor()]
+
+    fun getItemByIdOrNull(id: SkyBlockId): ItemStack? = cache.getOrPut(id.trySafe(::item)) {
+        val itemId = id.cleanId.uppercase()
+
+        if (itemId == UNKNOWN) {
+            return null
+        }
+
+        return RepoItemsAPI.getItemOrNull(itemId)
+    }
+
+    fun getItemById(id: SkyBlockId): ItemStack = getItemByIdOrNull(id) ?: ItemBuilder(Items.BARRIER) {
+        name("Unknown item: $id")
+    }
+
+    fun getPetByIdOrNull(id: SkyBlockId): ItemStack? = cache.getOrPut(id.trySafe(::pet)) {
+        val petId = id.cleanId.uppercase()
+
+        if (petId == UNKNOWN) {
+            return@getOrPut null
+        }
+
+        if (petId.contains(":")) {
+            val (petId, rarity) = petId.split(":")
+            val sbRarity = runCatching { SkyBlockRarity.valueOf(rarity) }.getOrNull()
+            val pet = runCatching {
+                sbRarity?.let { RepoPetsAPI.getPetAsItemOrNull(PetQuery(petId, it, 1)) }
+            }.getOrNull()
+            pet?.let { return@getOrPut it }
+        }
+
+        return@getOrPut SkyBlockRarity.entries.reversed().firstNotNullOfOrNull { skyBlockRarity ->
+            runCatching {
+                RepoPetsAPI.getPetAsItemOrNull(PetQuery(petId.substringBefore(":"), skyBlockRarity, 1))
+            }.getOrNull()
+        }
+    }
+
+    fun getPetById(id: SkyBlockId): ItemStack = getPetByIdOrNull(id) ?: ItemBuilder(Items.BARRIER) {
+        name("Unknown pet: $id")
+    }
+
+    fun getRuneByIdOrNull(id: SkyBlockId): ItemStack? = cache.getOrPut(id.trySafe(::rune)) {
+        val runeId = id.cleanId.uppercase()
+
+        if (runeId == UNKNOWN) {
+            return@getOrPut null
+        }
+
+        if (runeId.contains(":")) {
+            val (runeId, literalLevel) = runeId.split(":")
+            val level = literalLevel.toIntValue()
+            RepoRunesAPI.getRuneAsItemOrNull(runeId, level)?.let { return@getOrPut it }
+        }
+
+        for (i in 3 downTo 0) {
```

Comment:
allow the level to be nullable in the RepoRunesAPI to get the highest level of that rune. 

Make sure to do jvmoverloads to make it backwards compatible